src/sync/chase_lev.rs

```diff
@@ -254,6 +273,40 @@ impl<T> Deque<T> {
         }
     }
 
+    fn steal_half(&self) -> StealHalf<T> {
+        let guard = epoch::pin();
+
+        let t = self.top.load(Acquire);
+        fence(SeqCst);
+        let b = self.bottom.load(Acquire);
+
+        let size = b - t;
+        if size <= 0 {
+            return StealHalf::Empty;
+        }
+
+        let half = (size + 1) / 2;
+
+        unsafe {
+            let a = self.array.load(Acquire, &guard).unwrap();
+
+            let mut data = Vec::with_capacity(half as usize);
+            for i in t..t + half {
+                data.push(a.get(i));
+            }
+
+            if self.top.compare_and_swap(t, t + half, SeqCst) == t {
```

Comment:
@stjepang is right -- my bad.  This implementation of `steal_half()` is wrong, and I cannot imagine how to fix it.

The reason I thought it was correct is I only considered races among stealers.  As @stjepang said, `b` can be decremented beyond `t + half` before a `steal_half()` is successfully stealing from `t` to `t + half`.  That breaks deque's invariant.

Thank you pointing out that.  I think I should be more humble in front of concurrency...  :face_with_head_bandage:  I am closing this PR.

@fitzgen this PR is finally resolved, but negatively.  I will work more on designing a proper `steal_half()` in deque!